src/swdmt/redistricting/Redistrictor.java

```diff
@@ -68,18 +72,30 @@ public Region region() {
         int numberOfAugmentedDistricts
                 = theRegion.numberOfVoters() % numDistricts;
         Iterator<Location> locit = theRegion.locations().iterator();
+        
+        Location[] snakingLocations = new Location[theRegion.locations().size()];
```

Comment:
Why are the locations copied into an array?  (1) Why is a copy needed?  (2) Why an array rather than a list?

Reply:
We are relying on very specific ordering to ensure contiguity when locations are pulled in sequential order. I could've put the comparer in the Tree List, but I thought that added an unnecessarily tight coupling. I didn't want a future maintainer to change the Tree List ordering to break the generate method.

I used an array as opposed to a List because an array has less overhead.